File: 현준/해시/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        for (String part : participant) {
            if (map.containsKey(part)) {
                map.put(part, map.get(part) + 1);
            } else {
                map.put(part, 1);
            }
        }
        for (String comp : completion) {
            if (map.containsKey(comp)) {
                map.put(comp, map.get(comp) - 1);
            }
        }
        for (String key : map.keySet()) {
            if (map.get(key) > 0) {
                return key;
            }
        }
        return "";
    }
}

File: 계환/해시/완주하지 못한 선수.py
def solution(participant, completion):
    # 참여자명단에 있지만 완주자 명단에 없으면 완주하지 못한 선수

    # for i in participant:
    #     if i in completion:
    #         completion.remove(i)
    #     else:
    #         answer = i

    participant.sort()
    completion.sort()

    for i in range(len(completion)):
        if participant[i] != completion[i]:
            return participant[i]

    return participant[-1]
